cli/commands.go

```diff
@@ -198,7 +198,12 @@ func CreateCluster(c *cli.Context) error {
 				c.Int("port-auto-offset"),
 			)
 			if err != nil {
-				return fmt.Errorf("ERROR: failed to create worker node for cluster %s\n%+v", c.String("name"), err)
+				log.Printf("ERROR: failed to create worker node for cluster %s\n%+v", c.String("name"), err)
+				delErr := DeleteCluster(c)
+				if delErr != nil {
+					return delErr
+				}
+				os.Exit(1)
```

Comment:
A comment ahead: In general, I'm completely fine with this solution, the following is just a random thought :grin: 
So with `os.Exit(1)` we just introduced (probably already elsewhere in the code) a different way of exiting the CLI (instead of returning an error to the CLI framework). Couldn't we just construct the error message here from `err` and `delErr` and return it to the caller?
i.e. 
```Go
if err != nil {
  errorMessage := fmt.Errorf("ERROR: failed to create worker node for cluster %s\n%+v", c.String("name"), err)
  if delErr := DeleteCluster(c); delErr != nil {
    errorMessage = fmt.Errorf("%+v\n%+v", delErr)
  }
  return errorMessage
}
```
I didn't test this and am not sure if it works or if it even makes sense... as I said, just a random thought :+1:

Reply:
I agree with you on the point. I was just following the pattern that's already there in the code.  I'd prefer to deal with your suggestion in a different PR to improve the consistency in error handling.  Make sense?

Reply:
Sure thing, you can go ahead and merge this :+1:
There's an open issue for consistent logging and error handling anyways :grimacing: